### PR TITLE
bump library-go

### PR DESCRIPTION
**Pull request by mfojtik**
* openshift/library-go@3a6456f: resourcemerge: Allow removing annotations/labels
* openshift/library-go@423e202: factory: allow for post-start hooks for custom triggers
* openshift/library-go@7c9065e: implements a generic workload controller
* openshift/library-go@522f72d: go mod tidy
* openshift/library-go@07fd01b: adds precondition function
* openshift/library-go@34fe6fc: factory: remove support for runtime object in queue
* openshift/library-go@07eabbf: nsfinalizer: migrate controller to factory
* openshift/library-go@488e1fa: apiservice: migrate controller to factory
* openshift/library-go@38b127e: certrotation: migrate controller to factory
* openshift/library-go@3f9a525: configobserver: migrate controller to factory
* openshift/library-go@5f7087d: management: migrate controller to factory
* openshift/library-go@7f456e6: staleconditions: migrate controller to factory
* openshift/library-go@f17c0ee: status: migrate controller to factory
* openshift/library-go@892bfb1: resourcesync: adjust to queue key
* openshift/library-go@fa685ad: leaderelection: do not fatal on leader election lost
* openshift/library-go@c95f7e3: allows for setting an optional prefix for condition's type field for the generic workload ctrl
* openshift/library-go@3e49782: delegates creation of common informers down to workload controller
* openshift/library-go@a43f112: logging: properly set log level
* openshift/library-go@1add03a: resourcesynccontroller: fix unit test flakes

### CODE REVIEW

**Comment by mfojtik** [comment]
@deads2k pls review the certrotation changes

**Comment by mfojtik** [comment]
/retest

this is getting frustrating...

File: pkg/cmd/regeneratecerts/carry/kubecontrollermanager/certrotationcontroller/kubecontrollermanagercertrotation.go [hunk 1]
```diff
@@ -71,30 +74,21 @@ func NewCertRotationController(
...
-	if err != nil {
-		return nil, err
-	}
+	ret.recorder = eventRecorder
 	ret.certRotators = append(ret.certRotators, certRotator)
 
 	return ret, nil
 }
 
-func (c *CertRotationController) WaitForReady(stopCh <-chan struct{}) {
```
Comment (sttts) [inline pkg/cmd/regeneratecerts/carry/kubecontrollermanager/certrotationcontroller/kubecontrollermanagercertrotation.go]: where is this gone?

File: pkg/cmd/regeneratecerts/regenerate_certificates.go [hunk 2]
```diff
@@ -187,7 +187,7 @@ func (o *Options) Run() error {
 
 	// wait until the controllers are ready
 	kubeAPIServerCertRotationController.WaitForReady(ctx.Done())
-	kubeControllerManagerCertRotationController.WaitForReady(ctx.Done())
```
Comment (sttts) [inline pkg/cmd/regeneratecerts/regenerate_certificates.go]: where is this gone?

File: pkg/operator/certrotationcontroller/certrotationcontroller.go [hunk 3]
```diff
@@ -524,18 +511,15 @@ func (c *CertRotationController) WaitForReady(stopCh <-chan struct{}) {
 	if err := c.syncInternalLoadBalancerHostnames(); err != nil {
 		panic(err)
 	}
-
-	for _, certRotator := range c.certRotators {
```
Comment (sttts) [inline pkg/operator/certrotationcontroller/certrotationcontroller.go]: where is this gone?

File: pkg/cmd/regeneratecerts/carry/kubecontrollermanager/certrotationcontroller/kubecontrollermanagercertrotation.go (hunk 1 above)
Comment (mfojtik) [reply pkg/cmd/regeneratecerts/carry/kubecontrollermanager/certrotationcontroller/kubecontrollermanagercertrotation.go]: added back.

**Pull request merged**